From: Catalin Marinas <catalin.marinas@arm.com>
Date: Fri, 19 Feb 2016 14:28:58 +0000
Subject: arm64: User die() instead of panic() in do_page_fault()
Git-commit: 70c8abc28762d04e36c92e07eee2ce6ab41049cb
Patch-mainline: v4.6-rc1
References: bsc#1044339

The former gives better error reporting on unhandled permission faults
(introduced by the UAO patches).

Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/mm/fault.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm64/mm/fault.c
+++ b/arch/arm64/mm/fault.c
@@ -287,10 +287,10 @@ static int __kprobes do_page_fault(unsig
 
 	if (permission_fault(esr) && (addr < USER_DS)) {
 		if (get_thread_info(regs->sp)->addr_limit == KERNEL_DS)
-			panic("Accessing user space memory with fs=KERNEL_DS");
+			die("Accessing user space memory with fs=KERNEL_DS", regs, esr);
 
 		if (!search_exception_tables(regs->pc))
-			panic("Accessing user space memory outside uaccess.h routines");
+			die("Accessing user space memory outside uaccess.h routines", regs, esr);
 	}
 
 	/*
